pyqtgraph/parametertree/parameterTypes/action.py

```diff
@@ -94,3 +96,13 @@ class ActionParameter(Parameter):
     def activate(self):
         self.sigActivated.emit(self)
         self.emitStateChanged('activated', None)
+
+    def setValue(self, value, blocksignal=None):
```

Comment:
It would be good to document this new feature under the `Parameters` docstring block above. After `shortcut: str`, you can add a description of how `value` should be used.

Reply:
Finally getting back to this, sorry for the delay.

Pertinent question: is it appropriate to set the value to self.opts["value"]? I assume so, because that's the behavior of Parameter.setValue() that I'm overriding.

But I have an example script I'm using to check my work and if I have setValue store the value to self.opts the callable fires not once but twice-- ie. I get a popup when I click the button, and when I click OK & it closes I get another identical popup. Example script is attached for context.

[actionParameter_example_py.txt](https://github.com/pyqtgraph/pyqtgraph/files/9966770/actionParameter_example_py.txt)

Reply:
No worries, I haven't looked at it in a minute either. I think this logic might work a bit better:
```python
    def setValue(self, value, blockSignal=None):
        oldValue = self.value()
        if callable(oldValue) and oldValue is not value:
            disconnect(self.sigActivated, oldValue)
        if callable(value) and value is not oldValue:
            self.sigActivated.connect(value)
        return super().setValue(value, blockSignal)
```

This way, we always call `super().setValue()` and disconnect callable old values even when the new value isn't callable

Thanks for your example script! I tried it with this new implementation and it seemed to work OK: let me know what you think

Reply:
Well, I ran into an issue. I wanted to see your setValue() in action so I pasted it in, saved, then ran that example script using the python debugger. (action.py:76 is the first line of the method if it's pasted in at the end of the ActionParameterItem class.)

`kmfahey@Madeline:~/Workspace/pyqtgraph$ python -m pdb actionParameter_example.py 
> /home/kmfahey/Workspace/pyqtgraph/actionParameter_example.py(1)<module>()
-> import PyQt5
(Pdb) b /home/kmfahey/Workspace/pyqtgraph/pyqtgraph/parametertree/parameterTypes/action.py:76
Breakpoint 1 at /home/kmfahey/Workspace/pyqtgraph/pyqtgraph/parametertree/parameterTypes/action.py:76
(Pdb) c
qt.qpa.xcb: QXcbConnection: XCB error: 3 (BadWindow), sequence: 595, resource id: 24609331, major code: 40 (TranslateCoords), minor code: 0
The program finished and will be restarted
> /home/kmfahey/Workspace/pyqtgraph/actionParameter_example.py(1)<module>()
-> import PyQt5
(Pdb)`

The new setValue() was never called (assuming I pasted it in the right place). Did I err or is this issue resolved by some other code?

Reply:
Sorry for the long delay...

I'm a bit confused -- can you clarify what you mean by "the new setValue() was never called"?

Here's a different script that might better illustrate the code is working:
```python
import PyQt5
from pyqtgraph.Qt import QtWidgets
import pyqtgraph as pg
from pyqtgraph.parametertree import Parameter, ParameterTree, parameterTypes as ptypes


def a(x=5, y=6):
    QtWidgets.QMessageBox.information(None, "Hello World", f"X is {x}, Y is {y}")


params = Parameter.create(
    name='"a" parameters',
    type="group",
    children=[
        dict(name="x", type="int", value=5),
        dict(name="y", type="int", value=6),
        dict(
            name="Hello, world",
            type="action",
            value=lambda *args: pg.Qt.QtWidgets.QMessageBox.information(
                None, "hello world!", "hello world!"
            ),
        ),
    ],
)

def onChange(_param, _value):
   print(f"function is {_value}")

ap = params.child("Hello, world")
ap.sigValueChanged.connect(onChange)

ap.setValue(a)

app = pg.mkQApp()
tree = ParameterTree()
tree.setParameters(params)
tree.show()
pg.exec()


# ap.activate()

```